### PR TITLE
Add instant-run option

### PR DESCRIPTION
**Pull request by ItsProfessional**

### CODE REVIEW

**Comment by ItsProfessional** [comment]
@loserMcloser can you take a look at this?

File: completions/bash/waylogout [hunk 1]
```diff
@@ -58,6 +58,7 @@ _waylogout()
     --scaling
     --screenshots
     --scroll-sensitivity
+    --instant-run
```
Comment (loserMcloser) [inline completions/bash/waylogout]: The rest of these are in alphabetical order, can you move this line up to match?

File: completions/fish/waylogout.fish [hunk 2]
```diff
@@ -38,6 +38,7 @@ complete -c waylogout -l ring-selection-color        --description "Sets the col
 complete -c waylogout -l scaling                -s s --description "Image scaling mode."
 complete -c waylogout -l screenshots            -s S --description "Display a screenshot."
 complete -c waylogout -l scroll-sensitivity          --description "How fast selected action will change when scrolling with mouse/touch."
+complete -c waylogout -l instant-run                 --description "Instantly run actions on key press, without confirmation with enter key."
```
Comment (loserMcloser) [inline completions/fish/waylogout.fish:41]: Same comment here re alphabetical order

File: completions/zsh/_waylogout [hunk 3]
```diff
@@ -42,6 +42,7 @@ _arguments -s \
 	'(--scaling -s)'{--scaling,-s}'[Image scaling mode: stretch, fill, fit, center, tile, solid_color]:mode:(stretch fill fit center tile solid_color)' \
 	'(--screenshots -S)'{--screenshots,-S}'[Display a screenshot]' \
 	'(--scroll-sensitivity)'--scroll-sensitivity'[How fast selected action will change when scrolling with mouse/touch]:amount:' \
+	'(--instant-run)'--instant-run'[Instantly run actions on key press, without confirmation with enter key.]' \
```
Comment (loserMcloser) [inline completions/zsh/_waylogout]: alphabetical order again

File: input.c [hunk 4]
```diff
@@ -4,6 +4,22 @@
...
+    ts.tv_sec = msec / 1000;
+    ts.tv_nsec = (msec % 1000) * 1000000;
+
+    do {
+        res = nanosleep(&ts, &ts);
+    } while (res && errno == EINTR);
+
+    return res;
+}
+
```
Comment (loserMcloser) [inline input.c]: This seems like overkill:

1. I can't foresee any need to re-use this more general function in other places in the code.
2. You only ever call it with `msec` set to `1` so `ts.tv_sec` is always `0` and `ts.tv_nsec` is always `1000000`.
3. It seems very unlikely that `EINTR` will occur during this 1 msec delay. And even if it did, so what? So the `do` loop seems unnecessary.

Given all of the above, I think this whole block of code could be removed and replaced with a couple of lines in place of the actual `msleep(1)` call below.

File: input.c [hunk 5]
```diff
@@ -283,6 +299,12 @@ void waylogout_handle_key(struct waylogout_state *state,
 			if (action_iter->shortcut == keysym) {
 				state->selected_action = action_iter;
 				damage_state(state);
+
+        if(state->args.instant_run) {
+          msleep(1); // without this, the compositor crashes, or at least hyprland does.
+				  run_action(state->selected_action); // just returns if selected_action is NULL
+        }
+
```
Comment (loserMcloser) [inline input.c:290]: 1. The rest of this file uses tabs for indentation --- can you fix this code block to match?
2. I am on sway and I don't get any crash if I try it without the sleep. Do you get a compositor crash even if your associated command is only set to some `echo` command like in the sample command line in the README?
3. Replace the `msleep(1);` line with
````
			struct timespec ts = { .tv_sec = 0, .tv_nsec = 1000000 };
			nanosleep(&ts, NULL);
````

File: main.c [hunk 6]
```diff
@@ -998,6 +998,7 @@ static int parse_options(int argc, char **argv, struct waylogout_state *state,
 		LO_COMMAND_LOCK,
 		LO_COMMAND_SWITCH,
 		LO_COMMAND_SCROLL_SENSITIVITY,
+		LO_COMMAND_INSTANT_RUN,
```
Comment (loserMcloser) [inline main.c]: This should be called just `LO_INSTANT_RUN`. (lol, I did the same thing with `LO_COMMAND_SCROLL_SENSITIVITY`, should be just `LO_SCROLL_SENSITIVITY`. Not that it matters to the functionality of the code.)

**Review by loserMcloser — CHANGES_REQUESTED** [review]
Thanks for this! Can you fix these minor details?

**Pull request merged**